hw4/02-sessions.js

```diff
@@ -1,17 +1,38 @@
+
 const express = require('express');
 const session = require('express-session');
-const app = express();
-const port = process.env.PORT || 5001;
+const app     = express();
+const port    = process.env.PORT || 5001;
+
 
-// Add your code here
+app.use(session({secret: 'secret_key'}));
 
-// Use the express-session module
-app.use(/** ... */);
 
 app.get('/', (req, res) => {
-  // Add your code here
+  if (req.session.first_time) {
+    req.session.routes.push('/');
```

Comment:
This line assumes that the user launches the home page directly. This ignores the case when user visits the website for the first time through a different URL.
For eg. If user enters `localhost:5001/firstVisit` it would still save the history as `/` instead of `/firstvisit`